sdk/src/OpenTelemetry/Trace.hs

```diff
@@ -333,30 +335,35 @@ getTracerProviderInitializationOptions = getTracerProviderInitializationOptions'
 -}
 getTracerProviderInitializationOptions' :: (ResourceMerge 'Nothing any ~ 'Nothing) => Resource any -> IO ([SpanProcessor], TracerProviderOptions)
 getTracerProviderInitializationOptions' rs = do
-  sampler <- detectSampler
-  attrLimits <- detectAttributeLimits
-  spanLimits <- detectSpanLimits
-  propagators <- detectPropagators
-  processorConf <- detectBatchProcessorConfig
-  exporters <- detectExporters
-  builtInRs <- detectBuiltInResources
-  envVarRs <- mkResource . map Just <$> detectResourceAttributes
-  let allRs = mergeResources (builtInRs <> envVarRs) rs
-  processors <- case exporters of
-    [] -> do
-      pure []
-    e : _ -> do
-      pure <$> batchProcessor processorConf e
-  let providerOpts =
-        emptyTracerProviderOptions
-          { tracerProviderOptionsIdGenerator = defaultIdGenerator
-          , tracerProviderOptionsSampler = sampler
-          , tracerProviderOptionsAttributeLimits = attrLimits
-          , tracerProviderOptionsSpanLimits = spanLimits
-          , tracerProviderOptionsPropagators = propagators
-          , tracerProviderOptionsResources = materializeResources allRs
-          }
-  pure (processors, providerOpts)
+  disabled <- fmap (== "true") <$> lookupEnv "OTEL_SDK_DISABLED"
```

Comment:
I think we should use a _case-insensitive_ comparison here (see https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#boolean-value).
We should also fix the other occurrences of `fmap (== "true") <$> lookupEnv` in `OpenTelemetry.Exporter.OTLP.Span`.

Reply:
Good point, I misread that as case-_insensitive_ :sweat_smile: 

I agree the rest are also all wrong. If it's okay, I'll make a PR on top of this one fixing them all, so we remain consistent.

Reply:
I created a (draft) PR for this https://github.com/iand675/hs-opentelemetry/pull/153.